docs/cmd/changelogutils/changelog_utils.go

```
@@ -0,0 +1,288 @@
+	for _, release := range releases {
+		var releaseTag = release.GetTagName()
+		version, err := ParseVersion(releaseTag)
+		if err != nil {
+			return nil, err
+		}
+		var header string
+		// If byMinorVersion, we only want to include the release notes in the string and not the release header
+		if byMinorVersion {
+			header = fmt.Sprintf("##### %v\n", version.String())
+			version.LabelVersion, version.Patch, version.Label = 0, 0, ""
+		}
+		minorReleaseMap[*version] = minorReleaseMap[*version] + header + release.GetBody()
+	}
+
+	return minorReleaseMap, nil
+}
+
+// Performs processing to generate a map of release version to the release notes
+// This also pulls in open source gloo edge release notes and merges them with enterprise release notes
+// The returned map will be a mapping of minor releases (v1.5, v1.6) to their body, which will contain the release notes
+// for all the patches under the minor releases
+func MergeEnterpriseOSSReleases(enterpriseReleasesSorted, osReleases []*github.RepositoryRelease) (map[Version]string, error) {
+	var minorReleaseMap = make(map[Version]string)
+
+	openSourceReleases, err := ParseReleases(osReleases, false)
+	if err != nil {
+		return nil, err
+	}
+
+	for index, release := range enterpriseReleasesSorted {
+		var releaseTag = release.GetTagName()
+
+		version, err := ParseVersion(releaseTag)
+		if err != nil {
+			return nil, err
+		}
+
+		previousEnterprisePatch := GetPreviousEnterprisePatchVersion(enterpriseReleasesSorted, index)
+
+		// Get the Gloo OSS version that the Gloo enterprise version relies on
+		depVersion, err := GetOSSDependencyForEnterpriseVersion(version)
+		var OSSDescription string
+		if err == nil && previousEnterprisePatch != nil {
+			OSSDescription = GetOSDependencyPrefix(*depVersion, true)
+		}
+
+		//TODO: Implement a cache for gloo enterprise dependencies to reduce the amount of API calls
+		previousDepVersion, _ := GetOSSDependencyForEnterpriseVersion(previousEnterprisePatch)
```

add comment to explain why we are swallowing errors here?